source/agora/consensus/state/UTXOSet.d

```diff
@@ -49,18 +49,20 @@ abstract class UTXOCache
         Params:
             tx = the transaction
             height = local height of the block
+            slashed = whether the transaction is for slashing
 
     ***************************************************************************/
 
-    public void updateUTXOCache (const ref Transaction tx, Height height) @safe
+    public void updateUTXOCache (const ref Transaction tx, Height height,
+        bool slashed = false) @safe
     {
         import std.algorithm : any;
 
         // defaults to next block
         Height unlock_height = Height(height + 1);
 
         // for payments of frozen transactions, it will melt after 2016 blocks
-        if ((tx.type == TxType.Payment)
+        if ((tx.type == TxType.Payment && !slashed)
```

Comment:
You need to document these kinds of changes, it's very important.

For example:

```
If this is an unfreeze and slashing transaction, don't melt but instead set the unlock height to the next block as it will be sent to the commons budget.
```

Reply:
> You need to document these kinds of changes, it's very important.
> 
> For example:
> 
> ```
> If this is an unfreeze and slashing transaction, don't melt but instead set the unlock height to the next block as it will be sent to the commons budget.
> ```

Ah, I missed it although it's very important.